source/Government.cpp

```diff
@@ -32,31 +32,68 @@ using namespace std;
 
 namespace {
 	// Load ShipEvent strings and corresponding numerical values into a map.
-	void PenaltyHelper(const DataNode &node, map<int, double> &penalties)
+	void PenaltyHelper(const DataNode &node, map<int, Government::PenaltyEffect> &penalties)
 	{
+		auto loadPenalty = [&penalties](const DataNode &child, int eventType) -> void
+		{
+			double amount = child.Value(1);
+			// Reverse compatibility: transfer the provoke and atrocity events to special penalties.
+			Government::SpecialPenalty specialPenalty = eventType == ShipEvent::PROVOKE ?
+					Government::SpecialPenalty::PROVOKE : eventType == ShipEvent::ATROCITY ?
+					Government::SpecialPenalty::ATROCITY : Government::SpecialPenalty::NONE;
```

Comment:
I generally avoid stacking ternary operators.
```suggestion
			Government::SpecialPenalty specialPenalty = Government::SpecialPenalty::NONE;
			// Reverse compatibility: transfer the provoke and atrocity events to their respective
			// special penalties if one is not explicitly given to them.
			if(eventType == ShipEvent::PROVOKE)
				specialPenalty = Government::SpecialPenalty::PROVOKE;
			else if(eventType == ShipEvent::ATROCITY)
				specialPenalty = Government::SpecialPenalty::ATROCITY;
```